fix: Remove duplicate code and extract commons

devJaja opened the pull request:
I've successfully refactored duplicated code, centralized utilities, updated dependencies, and standardized API response handling. All task requirements are met.

Sendi0011 committed:
feat: implement Card Nav animation for landing header

- Add hover/active animations with card-style backgrounds
- Implement route-based active state detection using usePathname()
- Add comprehensive accessibility features (ARIA, keyboard nav)
- Create responsive mobile menu with slide animation
- Ensure light/dark theme compatibility
- Add smooth 300ms transitions for professional UX

Fixes #260

Sendi0011 committed:
Fix ESLint issues in HeaderLanding.tsx and switch to npm

hakymulla committed:
rental cancellation logic

hakymulla committed:
validate_cancellation

hakymulla committed:
test

Commit:
Add loading spinner and skeleton component

Pipemg25 committed:
Create readme-es.md

Pipemg25 committed:
Update readme-es.md

Ernestoo2 committed:
refactor: Implemented the redirect for the akkuea logo in the header

kimcascante commented:
@devJaja review this error please in CI

```Run go build ./...
  go build ./...
  shell: /usr/bin/bash -e {0}
# gin/common
Error: common/responses.go:5:2: "net/http" imported and not used
Error: Process completed with exit code 1.```

Ernestoo2 committed:
refactor: Added an authentication guard to settings page

Ernestoo2 committed:
refactor: Added an authentication guard to settings page

devJaja committed:
fix: Remove duplicate code and extract commons

devJaja committed:
fix: fix build and test

Clement-coder committed:
Improve rating UI with sharp design

Commit:
enhanced the not found page

kimcascante reviewed (approved):
Good work, thank you for your contribution

truthixify committed:
Add Getting Started section to README

truthixify committed:
fixed faulty emoji in getting started section in README

truthixify committed:
made some changes according to review

salazarsebas committed:
Merge branch 'main' into feature/card-nav-header-animation

salazarsebas reviewed (approved):
LGTM

aydamacink committed:
Optimize images using webp, while leaving png fallback for browsers that don't support webp

aydamacink committed:
undo changes in README

aydamacink committed:
add webp for herosection 1 and 2

aydamacink committed:
Add sizes to benefits and to hero images, adapt formatting

Clement-coder committed:
removed wrong routing

Clement-coder committed:
removed wrong routing

salazarsebas committed:
docs: prepare the folder for issues

salazarsebas committed:
docs: prepare the folder for issues

salazarsebas committed:
docs: prepare the folder for issues

zintarh committed:
feat: implement and integrate pagination into existing list pages

zintarh committed:
fix: conflict

kimcascante commented:
@devJaja please review conflicts

Sendi0011 committed:
Fix nav cards: unify all to white background

Sendi0011 committed:
Fix conflicts: resolve conflict

kimcascante reviewed (changes requested):
Please resolve the conflicts @devJaja

devJaja committed:
Merge branch 'main' into eliminate-duplicate-code

salazarsebas commented:
Hey @devJaja , please resolve the errors on the workflow

devJaja committed:
Merge branch 'main' into eliminate-duplicate-code

devJaja committed:
Merge branch 'main' into eliminate-duplicate-code

devJaja committed:
fix(tailwind): resolve TypeScript error and improve DX

This commit addresses a TypeScript error in the tailwind.config.ts file by replacing a require statement with an import, aligning with modern ES module standards. It also enhances the development experience by adding a workspaces field to the root package.json, ensuring consistent dependency installation across the project. Additionally, a missing dependency has been added to the assistant package, and a new .eslintrc.json file has been created to maintain code quality.

devJaja committed:
fix(deps): resolve merge conflict and update dependencies

devJaja committed:
chore: merge changes from remote

devJaja committed:
fix: Next.js Frontend CI / build fix

devJaja committed:
fix: Next.js Frontend CI / build fix

Sendi0011 committed:
Merge branch 'main' into feature/card-nav-header-animation

Wetshakat committed:
Merge branch 'main' into Create-Loading-Component-#266

salazarsebas commented on docs/README.md:
```diff
@@ -1,14 +1,18 @@
 
 ---
 
+Issue 295
```
Hey, please don't add this 3 lines

devJaja replied:
> Hey, please don't add this 3 lines

Gm
please which of the lines am I removing. 9,10,and 11 ?

salazarsebas replied:
16, 17 and 18

Marvelousmicheal committed:
feat:Add i18n support for multiple languages

Marvelousmicheal committed:
feat:Add i18n support for multiple languages

akintewe committed:
Task #165: Add simple manual authentication tests

- Add cURL script to test POST /auth/register with valid and invalid inputs
- Add cURL script to test POST /auth/login with correct and incorrect credentials
- Add temporary protected route GET /protected for testing
- Document test cases in README.md for reproducibility
- Use simple manual testing approach as requested by moderator

Fulfills Task #165 requirements without over-engineering.

Marvelousmicheal committed:
feat:Add i18n support for multiple languages

Marvelousmicheal committed:
feat:Add i18n support for multiple languages

Marvelousmicheal committed:
fix: cli fix

Marvelousmicheal committed:
fix: cli fix

devJaja committed:
Merge branch 'main' into eliminate-duplicate-code

xJeffx23 committed:
Merge pull request #279 from Sendi0011/feature/card-nav-header-animation

feat: implement Card Nav animation for landing header

xJeffx23 committed:
Merge pull request #291 from truthixify/fix-261

Improved README with "Getting Started" Section

xJeffx23 committed:
Merge pull request #339 from zintarh/pagination

feat: add reusable pagination system and integrate across list pages

Marvelousmicheal committed:
fix: cli fix

xJeffx23 committed:
Merge pull request #352 from akintewe/feature/task-165-simple-manual-tests

Task #165: Add simple manual authentication tests

salazarsebas committed:
Merge branch 'main' into fix-269-optimize-images-with-modern-formats

salazarsebas committed:
Merge pull request #351 from Marvelousmicheal/feat/intl-language-switcher

feat: Implement i18n and Language Switcher

Bosun-Josh121 committed:
feat: Advanced tipping features and analytics

salazarsebas committed:
Merge pull request #359 from Bosun-Josh121/feat/advanced-tipping-features-analytics

feat:  Advanced tipping features and analytics

diegoveme committed:
docs: add documentation structure for issue 311

salazarsebas committed:
Merge pull request #360 from diegoveme/docs/issue-311

docs: add documentation structure for issue 311

SebasHidalgo committed:
docs: document the agent types and specializations

aydamacink committed:
Merge branch 'main' into fix-269-optimize-images-with-modern-formats

KAmaL-senpai committed:
Add training pipeline diagram for issue 335

KAmaL-senpai committed:
Add training pipeline diagram for issue 335

KAmaL-senpai committed:
Add training pipeline diagram for issue 335

KAmaL-senpai committed:
Add training pipeline diagram for issue 335

KAmaL-senpai committed:
Add training pipeline diagram for issue 335

KAmaL-senpai committed:
Add training pipeline diagram for issue 335

KAmaL-senpai committed:
Add training pipeline diagram for issue 335

Wetshakat committed:
Merge branch 'main' into Create-Loading-Component-#266

kimcascante committed:
Merge pull request #361 from SebasHidalgo/docs/agent-types-and-specializations

docs: document the agent types and specializations

kimcascante committed:
Merge pull request #362 from KAmaL-senpai/Document-the-Training-Pipeline-335

Document the training pipeline 335

salazarsebas committed:
chore: minimal change on docs

salazarsebas committed:
chore: minimal change on docs

salazarsebas committed:
chore: minimal change on docs

salazarsebas committed:
chore: minimal change on docs

salazarsebas committed:
chore: minimal change on docs

salazarsebas commented:
Hey @devJaja , please resolve the conflicts. Apologies for the inconvenience, but having so many issues causes a lot of conflicts.

salazarsebas committed:
Merge pull request #285 from Wetshakat/Create-Loading-Component-#266

Add loading spinner and skeleton component

salazarsebas committed:
Merge branch 'main' into fix-269-optimize-images-with-modern-formats

xJeffx23 committed:
Merge pull request #292 from aydamacink/fix-269-optimize-images-with-modern-formats

Fix: Optimize Images with Modern Formats

Pipemg25 committed:
Merge branch 'akkuea:main' into main

Pipemg25 committed:
Update readme-es.md

salazarsebas committed:
Merge pull request #363 from Pipemg25/main

263

devJaja committed:
fix:conflict fixes

devJaja committed:
Merge branch 'eliminate-duplicate-code' of https://github.com/devJaja/akkuea into eliminate-duplicate-code

devJaja committed:
fix:conflict fixes

devJaja committed:
Merge branch 'eliminate-duplicate-code' into main after resolving conflicts

devJaja committed:
fix: Resolve merge conflicts from eliminate-duplicate-code into main

devJaja committed:
fix: CI/CD fixes
